core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -56,76 +55,197 @@ trait ConfigConvert[T] {
+  }
+
+  def fromNonEmptyStringReaderOpt[T](fromF: String => Option[T])(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromNonEmptyStringReader[T](optF(fromF))
+  }
+
+  @deprecated(message = "The usage of Try has been deprecated. Please use fromStringConvert instead", since = "0.6.0")
+  def stringConvert[T](fromF: String => Try[T], toF: T => String): ConfigConvert[T] =
+    fromStringConvert[T](fromF andThen tryToEither, toF)
+
+  def fromStringConvert[T](fromF: String => Either[ConfigReaderFailure, T], toF: T => String): ConfigConvert[T] = new ConfigConvert[T] {
+    override def from(config: ConfigValue): Either[ConfigReaderFailures, T] = stringToEitherConvert(fromF)(config)
     override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(toF(t))
   }
 
-  def nonEmptyStringConvert[T](fromF: String => Try[T], toF: T => String)(implicit ct: ClassTag[T]): ConfigConvert[T] = {
-    stringConvert(ensureNonEmpty(ct)(_).flatMap(fromF), toF)
+  def fromStringConvertTry[T](fromF: String => Try[T], toF: T => String)(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromStringConvert[T](tryF(fromF), toF)
+  }
+
+  def fromStringConvertOpt[T](fromF: String => Option[T], toF: T => String)(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromStringConvert[T](optF(fromF), toF)
+  }
+
+  @deprecated(message = "The usage of Try has been deprecated. Please use fromNonEmptyStringConvert instead", since = "0.6.0")
+  def nonEmptyStringConvert[T](fromF: String => Try[T], toF: T => String)(implicit ct: ClassTag[T]): ConfigConvert[T] =
+    fromNonEmptyStringConvert[T](fromF andThen tryToEither[T], toF)
+
+  def fromNonEmptyStringConvert[T](fromF: String => Either[ConfigReaderFailure, T], toF: T => String)(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromStringConvert[T](ensureNonEmpty(ct)(_).right.flatMap(fromF), toF)
+  }
+
+  def fromNonEmptyStringConvertTry[T](fromF: String => Try[T], toF: T => String)(implicit ct: ClassTag[T]) = {
+    fromNonEmptyStringConvert[T](tryF(fromF), toF)
+  }
+
+  def fromNonEmptyStringConvertOpt[T](fromF: String => Option[T], toF: T => String)(implicit ct: ClassTag[T]) = {
+    fromNonEmptyStringConvert[T](optF(fromF), toF)
   }
 
   private[pureconfig] trait WrappedConfigConvert[Wrapped, SubRepr] extends ConfigConvert[SubRepr]
 
   private[pureconfig] trait WrappedDefaultValueConfigConvert[Wrapped, SubRepr <: HList, DefaultRepr <: HList] extends WrappedConfigConvert[Wrapped, SubRepr] {
-    final def from(config: ConfigValue): Try[SubRepr] =
-      Failure(
-        new UnsupportedOperationException("Cannot call 'from' on a WrappedDefaultValueConfigConvert."))
-    def fromWithDefault(config: ConfigValue, default: DefaultRepr): Try[SubRepr] = config match {
-      case co: ConfigObject =>
-        fromConfigObject(co, default)
-      case other =>
-        Failure(WrongTypeException(config.valueType().toString))
+    final def from(config: ConfigValue): Either[ConfigReaderFailures, SubRepr] =
```

Comment:
I think it was me who made `WrappedDefaultValueConfigConvert` implementing `ConfigConvert`, but, in retrospect, I agree with you that it doesn't make much sense. This `from` method is pretty much inaccessible code.